### PR TITLE
Finished 5th hw

### PR DESCRIPTION
**Pull request by yaourtv**
### About Mentor class:
`notifications` method uses timestamps to check new notifications. It means that mentor has no notifications before they call this method. 
- **Why this solution is bad?**
Well, if you call `read_notifications!` before `notifications` nothing would be changed.
- **How it could be fixed?**
I had one more idea that I still not implemented: create MentorList file with all Mentor class instances. When `Student#submit_homework` method is called it also calls Notifications class method (smth like `send_notifications`), which goes through all MentorList, check `subscription[ ]` and add new element in `notifs[ ]`.
- **Why not fixed yet?**
This app would create file which is not good if we need to store all variables and instance inside session.
**UPD:** check comment after commit cff44f8b429aec0ed955efd1569cdff39ddcb42e

### About Homework class:
- What is `mark`? Why -1, 0, 1?
Check `Mentor#check_homework` and comments in Homework class. TL;DR it describes status of the homework since we don't need the exact mark.

### About Human class:
This class has only one variable. So in this task we don't really need it. Without this class our Mentor and Student would get bigger (one more var, two more lines and contain more context) but it's not critical and I guess would give better performance in our task at the cost of scaling.

### CODE REVIEW

**Comment by yaourtv** [comment]
Was digging around Notification/Subscriber/EventListener patterns and found [Observer pattern](https://medium.com/@mitchocail/observer-pattern-in-ruby-e80ac3c1dac7). Rewrote some code in Mentor and Student classes. Now when Mentor subscribes to Student it calls `Student#get_subscriber` method and stores Mentor class instance. Now notifications work as they should: subscriber get new notification at the exact moment student submit their homework.

File: HW_05/Egor_Alekseychik/mentor.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+  def subscribe_to_student(student)
+    return unless already_subscribed?(student)
+
+    subscription << { student: student, time: Time.now.iso8601(6) }
+    student.get_subscriber(self)
+  end
+
+  def notifications
+    customized_output(notifs)
```
Comment (labs-ruby) [inline HW_05/Egor_Alekseychik/mentor.rb]: Not sure whether output formatting is `Mentor` class responsibility.
Consider using something like that
```ruby
Notifications::Formatter.new(notifications).print!
```

File: HW_05/Egor_Alekseychik/mentor.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  def findhw(student, pr_title)
+    student.homeworks.each do |hw|
+      return hw.source_code if hw.pr_title == pr_title
+    end
+  end
+
+  def customized_output(notifs)
+    out = []
+    notifs.each do |note|
```
Comment (labs-ruby) [inline HW_05/Egor_Alekseychik/mentor.rb]: Use `Array#select`

**Comment by yaourtv** [comment]
### About Notifications class
I don't think that we need to use module namespaces to have classes that do only one job. And it might be hard to test so I decided to create just a class that belongs to Mentor and allows Student to work with it (in this case Mentor do not modify notifications and student don't create them)

**Pull request merged**